### DISCUSSION TITLE
@ObservableState change doesn't update View

### DISCUSSION THREAD

**Discussion by marisol-develop** [pointfreeco/swift-composable-architecture#3122] [Q&A] [answered]
Hello there 👋 

Currently, I'm using TCA version of 1.10.4, and Xcode minimum deployments of 16.0.

I have some problem when updates view with ObservableState.

1️⃣ Changes in ObservableState do not trigger view updates intermittently.

This issue does not occur every time, but intermittently fails to trigger view updates.
Additionally, using `_printChanges` also prints nothing even when the view updated as expected.

Wrapping a subview with another WithPerceptionTracking inside WithPerceptionTracking resolves the issue. I want to understand why I need to wrap it with WithPerceptionTracking twice.

2️⃣ WithPerceptionTracking warning occurs even when the view is wrapped with `WithPerceptionTracking`.

When the view appears, 12 purple warning are shown. (`Perceptible state was accessed but is not being tracked. Track changes to state by wrapping your view in a 'WithPerceptionTracking' view.`)

I wonder if I used WithPerceptionTracking incorrectly.
Since this code is from a company project, I cannot share the whole project file, but I will provide a brief snippet.

FYI, that UserInfoEditView is pushed by NavigationLink in Vanilla SwiftUI. Could that potentially have an impact?

Thank you in advance for your help! 

```
struct UserInfoEditView: View {
    @Perception.Bindable var store: StoreOf<UserInformationEditFeature>
    ...
    var body: some View {
        WithPerceptionTracking {
            ScrollViewReader { proxy in
                ScrollView {
                    ...

                    // 🤔 It works well when I wrap this VStack with WithPerceptionTracking again
                    VStack {
                        Button {
                            store.send(.nameTooltipTapped)
                        } label: {
                            Text("Button")
                                .foregroundStyle(store.showNameTooltip ? Color(.gray500) : Color(.red)) // ❌ It is not updated
                        }
                        ...
                    } // VStack
                } // ScrollView
            } // ScrollViewReader
        }
    }
}
```

```
@Reducer
struct UserInfoEditFeature {
    @ObservableState
    struct State: Equatable {
        var showNameTooltip: Bool = false
        ...
    }

    enum Action: BindableAction {
        case nameTooltipTapped
        ...
    }

    var body: some ReducerOf<Self> {
        BindingReducer()

        Reduce<State, Action> { state, action in
            case .nameTooltipTapped:
                state.showNameTooltip = true
                return .none
        }
    }
}
```

**Comment by kevinanderson7** [comment, marked as answer]
Hey @marisol-develop, I think this might answer your question. `ScrollViewReader` is an example of one of the views that requires this:

https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/observationbackport#Lazy-view-closures

**Reply by marisol-develop** [reply]
Thanks for your help @kevinanderson7 !👍
